src/main/java/org/stellar/sdk/StrKey.java

```diff
@@ -130,6 +196,10 @@ public static String encodeSha256Hash(byte[] data) {
     }
 
     protected static byte[] decodeCheck(VersionByte versionByte, char[] encoded) {
+        if (encoded.length == 0) {
+            throw new IllegalArgumentException("Encoded char array cannot be empty.");
+        }
+
         byte[] bytes = new byte[encoded.length];
         for (int i = 0; i < encoded.length; i++) {
             if (encoded[i] > 127) {
```

Comment:
Not relevant to this PR, but why is this necesary? `base32Encoding.decode` should be taking care of this, shouldn't it?

Reply:
Maybe this is needed due to `guava` not handling it properly ...